include/CLI/TypeTools.hpp

```diff
@@ -607,6 +719,160 @@ bool lexical_assign(const std::string &input, T &output) {
+        strings[0], std::get<0>(output));
+    return retval;
+}
+
+/// Conversion for single element tuple and single defined type
+template <class T,
+          class XC,
+          enable_if_t<type_count<T>::value == 1 && is_tuple_like<T>::value && !is_tuple_like<XC>::value,
+                      detail::enabler> = detail::dummy>
+bool lexical_conversion(const std::vector<std ::string> &strings, T &output) {
+    static_assert(type_count<T>::value == type_count<XC>::value,
+                  "when using converting to tuples different cross conversion are not possible");
+
+    bool retval = lexical_assign<typename std::tuple_element<0, T>::type, XC>(strings[0], std::get<0>(output));
+    return retval;
+}
+
+/// conversion for two element tuple
+template <class T, class XC, enable_if_t<type_count<T>::value == 2, detail::enabler> = detail::dummy>
+bool lexical_conversion(const std::vector<std ::string> &strings, T &output) {
+    static_assert(type_count<T>::value == type_count<XC>::value,
+                  "when using converting to tuples different cross conversion are not possible");
+
+    bool retval = lexical_cast(strings[0], std::get<0>(output));
+    if(strings.size() > 1) {
+        retval &= lexical_cast(strings[1], std::get<1>(output));
+    }
+    return retval;
+}
+
+/// conversion for three element tuple
+template <class T, class XC, enable_if_t<type_count<T>::value == 3, detail::enabler> = detail::dummy>
+bool lexical_conversion(const std::vector<std ::string> &strings, T &output) {
+    static_assert(type_count<T>::value == type_count<XC>::value,
+                  "when using converting to tuples different cross conversion are not possible");
+
+    bool retval = lexical_cast(strings[0], std::get<0>(output));
+    if(strings.size() > 1) {
+        retval &= lexical_cast(strings[1], std::get<1>(output));
+    }
+    if(strings.size() > 2) {
+        retval &= lexical_cast(strings[2], std::get<2>(output));
+    }
+    return retval;
+}
+
+/// conversion for four element tuple
+template <class T, class XC, enable_if_t<type_count<T>::value == 4, detail::enabler> = detail::dummy>
+bool lexical_conversion(const std::vector<std ::string> &strings, T &output) {
+    static_assert(type_count<T>::value == type_count<XC>::value,
+                  "when using converting to tuples different cross conversion are not possible");
+
+    bool retval = lexical_cast(strings[0], std::get<0>(output));
+    if(strings.size() > 1) {
+        retval &= lexical_cast(strings[1], std::get<1>(output));
+    }
+    if(strings.size() > 2) {
+        retval &= lexical_cast(strings[2], std::get<2>(output));
+    }
+    if(strings.size() > 3) {
+        retval &= lexical_cast(strings[3], std::get<3>(output));
+    }
+    return retval;
+}
+
+/// conversion for five element tuple
+template <class T, class XC, enable_if_t<type_count<T>::value == 5, detail::enabler> = detail::dummy>
```

Comment:
It seems like there should be a recursive way to do this.

Reply:
For some parts I think it is doable with a compile time recursion, but it wasn't clear to me how to transmit the runtime state along with the compile time recursion and keep everything sane.  Would love it if you or anyone knew how to do that but I was not seeing any clear solutions to get around the limit of 5 in a nice way.  Not sure how much value there is in supporting 6 element tuples either.